content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,437 @@
+
+```php
+ddappsec.log_level=‘debug’
+ddappsec.helper_extra_args=‘--log_level=debug’
+ddappsec.helper_log_file=‘/tmp/helper.log’
+```
+
+The extension outputs logs to the default `php_error` log file. If there are no logs in the file, add the following to the `.ini` file:
+
+```php
+ddappsec.log_file=’tmp/extension.log’
+```
+
+### Installation fails to find PHP
+If the installation script is unable to find the correct PHP version, you can set the `--php-bin` to the PHP binary location, for example:
+
+```
+$ php dd-library-php-setup.php --php-bin /usr/bin/php7.4 --tracer-version=latest --appsec-version=latest
+```
+### Connection to helper failed
+If the Application Security extension is unable to communicate with the helper process, you will see the following log warning:
+
+```
+PHP Warning:  Unknown: [ddappsec] Connection to helper failed and we are not going to attempt to launch it: dd_error
+```
+
+The warning could be followed by one of these error messages:
+
+```
+PHP Warning:  Unknown: [ddappsec] Could not open lock file /tmp/ddappsec.lock: Permission denied in Unknown on line 0
+```
+```
+PHP Warning:  Unknown: [ddappsec] Call to bind() failed: Permission denied
+```
+```
+PHP Warning:  Unknown: [ddappsec] Failed to unlink /tmp/ddappsec.sock: Operation not permitted
+```
+
+This indicates that the lock file or socket file used by the extension has invalid permissions, or the user executing the PHP process does not have write access to the `tmp` directory. 
+
+If the lock file or socket file has invalid permissions, you can either delete them and restart Apache/FPM or adjust the `user:group` to match the one used by Apache/FPM, for example, `www-data`.
+
+If the user doesn’t have write access to the tmp directory, you can change the location of the lock file and socket file by modifying the following settings in the extension’s `.ini` file:
+
+```
+ddappsec.helper_socket_path = /<directory with compatible permissions>/ddappsec.sock
+ddappsec.helper_lock_path = /<directory with compatible permissions>/ddappsec.lock
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
+#### Installing the C toolchain
+
+CGO requires the following:
+
+- The `gcc` compiler for the target `GOOS` and `GOARCH`.
+- The C library headers.
+
+To install those libraries, use the command based on your OS:
+
+| OS                   | Installation Command          |
+|----------------------|-------------------------------|
+| Debian, Ubuntu       | $ apt install gcc libc6-dev   |
+| Alpine               | $ apk add gcc musl-dev	       |
+| RHEL, CentOS, Fedora | $ yum install gcc glibc-devel |
+| macOS	               | xcode-select --install        |
+
```

Comment:
"Installing the C toolchain" is already in the Setup and Configuration, remove this?

Reply:
+1 to removing if there isn't add't context here